velox/functions/lib/LikeFunctionUtils.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LikeFunctionUtils.h"
+
+#include <optional>
+
+#include "velox/expression/EvalCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/lib/ArrayBuilder.h"
+#include "velox/type/StringView.h"
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+using ::facebook::velox::exec::EvalCtx;
+using ::facebook::velox::exec::Expr;
+using ::facebook::velox::exec::VectorFunction;
+using ::facebook::velox::exec::VectorFunctionArg;
+
+// This function checks if the string 'pattern' has wildcard characters; a
+// wildcard character could either be '_' or '%'. If wildcard characters are
+// detected in the string 'pattern', the function returns true; else it returns
+// false.
+bool checkWildcardInPattern(const std::string& pattern) {
+  size_t patternLength = pattern.size();
+  for (size_t patternIndex = 0; patternIndex < patternLength; patternIndex++) {
+    if ((pattern[patternIndex] == '%') || (pattern[patternIndex] == '_')) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// This function is called if the string 'pattern' has no wildcard characters; a
```

Comment:
I feel we should use the word "fixed" pattern instead of your usage of "exact". That would make the logic very clear.